src/cryptoadvance/specter/util/mnemonic.py

```diff
@@ -30,14 +34,52 @@ def generate_mnemonic(strength=256, language_code="en") -> str:
     "tomorrow question cook lend burden bone own junior stage square leaf father edge decrease pipe tired useful junior calm silver topple require rug clock"
 
     :param strength: 256 (default) will give you 24 words. 128 will result in 12 words
-    :param language_code: only "en", "es", "fr" and "it" are supported. If you use an unsupported one, it'll fallback to english
+    :param language_code: only "en", "es", and "it" are supported. If you use an unsupported one, it'll fallback to English.
     """
     mnemo = initialize_mnemonic(language_code)
     return mnemo.generate(strength=strength)
 
 
-def validate_mnemonic(words):
-    # We cannot assume the mnemonic will be in the same language currently active
-    #   in the UI (e.g. a Spanish user is likely to have an English mnemonic).
-    mnemo = initialize_mnemonic(Mnemonic.detect_language(words))
-    return mnemo.check(words)
+def get_language(mnemonic: str) -> str:
+    try:
+        supported_language_found = False
+        language = Mnemonic.detect_language(mnemonic)
+        words_as_list = mnemonic.split()
+        # If we get French as language we have to double check whether it is not really an English mnemonic because of overlaps in the wordlists
+        if language == "french":
+            count = 0
+            for word in words_as_list:
+                if word in WORDLIST:
+                    count += 1
+            if count == 12:
+                language = "english"
+        for key, value in MNEMONIC_LANGUAGES.items():
+            if value == language:
+                supported_language_found = True
+                return language
+        if supported_language_found == False:
+            raise SpecterError(
+                f"The language {language.capitalize()} is not supported."
+            )
+    except ConfigurationError as ce:
+        raise SpecterError(str(ce))
+
+
+def validate_mnemonic(mnemonic: str) -> bool:
```

Comment:
In python, you can call a function with a positional argument as it would be a keyword-argument. Like this:
```
>>> def testf(words):
...     print(words)
... 
>>> testf("bla")
bla
>>> testf(words="bla")
bla
>>> testf(mnemonic="bla")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: testf() got an unexpected keyword argument 'mnemonic'
>>> 
```

You changed it from `words` to `mnemonic` and so we get:

```
[   INFO] in       _internal: 127.0.0.1 - - [14/Apr/2022 10:45:39] "GET /static/qr/qr-scanner.min.js HTTP/1.1" 200 -
[  ERROR] in      controller: Uncaught exception: validate_mnemonic() got an unexpected keyword argument 'words'
[  ERROR] in      controller: Traceback (most recent call last):
  File "/tmp/cirrus-ci-build/.env/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/tmp/cirrus-ci-build/.env/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/tmp/cirrus-ci-build/.env/lib/python3.8/site-packages/flask_login/utils.py", line 269, in decorated_view
    return func(*args, **kwargs)
  File "/tmp/cirrus-ci-build/src/cryptoadvance/specter/server_endpoints/devices.py", line 207, in new_device_mnemonic
    if not validate_mnemonic(words=request.form["mnemonic"]):
TypeError: validate_mnemonic() got an unexpected keyword argument 'words'
```
https://cirrus-ci.com/task/6263390917099520?logs=cypress#L1049-L1058

So check for all calls of that function and fix the calls.